pkg/arvo/app/chat-cli.hoon

```diff
@@ -0,0 +1,1111 @@
+    ::
+      [%create path =(unit glyph)]                  ::  create chat
+      [%join target =(unit glyph)]                  ::  join target
+      [%leave target]                               ::  nuke target
+    ::
+      [%bind glyph target]                          ::  bind glyph
+      [%unbind glyph]                               ::  unbind glyph
+      [%what (unit $@(char target))]                ::  glyph lookup
+    ::
+      [%settings ~]                                 ::  show active settings
+      [%set term]                                   ::  set settings flag
+      [%unset term]                                 ::  unset settings flag
+      [%width @ud]                                  ::  adjust display width
+      [%timezone ? @ud]                             ::  adjust time printing
+    ::
+      [%select $@(rel=@ud [zeros=@u abs=@ud])]      ::  rel/abs msg selection
+      [%chats ~]                                    ::  list available chats
+      [%help ~]                                     ::  print usage info
+  ==                                                ::
+::
++$  move  [bone card]
++$  card
+  $%  [%diff %sole-effect sole-effect:sole-sur]
+      [%poke wire dock out-action]
+      [%peer wire dock path]
+  ==
+::
++$  out-action
+  $%  [%chat-action chat-action]
+      [%chat-view-action chat-view-action]
+      [%chat-hook-action chat-hook-action]
+  ==
+--
+::
+|_  [=bowl:gall state]
+++  this  .
+::  +prep: setup & state adapter
+::
+++  prep
+  |=  old=(unit state)
+  ~&  %chat-cli-prep
+  ?^  old
+    [~ this(+<+ u.old)]
+  :-  [connect ~]
+  %_  this
+    audience  [[our-self /] ~ ~]
+    settings  (sy %showtime %notify ~)
+    width  80
+  ==
+::  +connect: connect to the chat-store
+::
+++  connect
+  ^-  move
+  [ost.bowl %peer /chat-store [our-self %chat-store] /all]
+::  +true-self: moons to planets
+::
+++  true-self
+  |=  who=ship
+  ^-  ship
+  ?.  ?=(%earl (clan:title who))  who
+  (sein:title our.bowl now.bowl who)
+++  our-self  (true-self our.bowl)
+::  +target-to-path: prepend ship to the path
+::
+++  target-to-path
+  |=  target
+  path
+  ::TODO
+  :: [(scot %p ship) path]
+::  +path-to-target: deduces a target from a mailbox path
+::
+++  path-to-target
+  |=  =path
+  ^-  target
+  ?.  ?=([@ @ *] path)
+    ::TODO  but then doing target-to-path won't get us the same path...
+    [our-self path]
+  =+  who=(slaw %p i.path)
+  ?~  who  [our-self path]
+  [u.who path]
+::  +poke-noun: debug helpers
+::
+++  poke-noun
+  |=  a=*
+  ^-  (quip move _this)
+  ?:  ?=(%connect a)
+    [[connect ~] this]
+  [~ this]
+::  +poke-sole-action: handle cli input
+::
+++  poke-sole-action
+  |=  act=sole-action:sole-sur
+  ^-  (quip move _this)
+  ?.  =(bone.cli ost.bowl)
+    ~|(%strange-sole !!)
+  (sole:sh-in act)
+::  +peer: accept only cli subscriptions from ourselves
+::
+++  peer
+  |=  =path
+  ^-  (quip move _this)
+  ?.  (team:title our-self src.bowl)
```

Comment:
This is great. Can you make it an assert with ?>

Reply:
This pattern is necessary to limit the `~|` error trace message to the specific condition. If you first `~|` and then `?>`, the message will be applied to failing assertions *and* any subsequent error. It'd be nice if we had a rune for this, but it's a fine pattern as-is.

Reply:
Resolving as wontfix.